app/controllers/programming_groups_controller.rb

```diff
@@ -65,21 +81,42 @@ def create
     end
   end
 
+  def update
+    myparams = programming_group_params2
+    myparams[:users] =
+      @programming_group.programming_group_memberships.where(id: myparams[:programming_group_membership_ids].compact_blank).includes(:user).map(&:user)
+    myparams.delete(:programming_group_membership_ids)
+    update_and_respond(object: @programming_group, params: myparams)
+  end
+
+  def destroy
+    authorize!
+    destroy_and_respond(object: @programming_group)
+  end
+
   private
 
   def authorize!
-    authorize(@programming_group)
+    authorize(@programming_group || @programming_groups)
   end
 
   def programming_group_params
     params.require(:programming_group).permit(:programming_partner_ids)
   end
 
+  def programming_group_params2
+    params[:programming_group].permit(:id, :exercise_id, programming_group_membership_ids: []) if params[:programming_group].present?
+  end
+
   def set_exercise_and_authorize
     @exercise = Exercise.find(params[:exercise_id])
     authorize(@exercise, :implement?)
   end
 
+  def set_programming_group
+    @programming_group = ProgrammingGroup.find(params[:id])
+  end
```

Comment:
Let's ensure we are always authorizing:

```suggestion
  def set_programming_group_and_authorize
    @programming_group = ProgrammingGroup.find(params[:id])
    authorize!
  end
```

Please remove the "manual" `authorize!` from all `MEMBER_ACTIONS`.